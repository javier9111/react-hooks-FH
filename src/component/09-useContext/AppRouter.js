import React from "react";

import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { LoginScreen } from "./LoginScreen";
import { HomeScreen } from "./HomeScreen";
import { NavBar } from "./NavBar";
export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Switch>
            {/* exact sirve para redirigir exactament como lo indica el path asi se evita que la expresion regular tome todo en lugar de lo que solo se le presente ej. "/" /login */}
            <Route exact path="/" component={HomeScreen} />
            <Route exact path="/about" component={AboutScreen} />
            <Route exact path="/login" component={LoginScreen} />

            {/*Este es para crear un default donde las rutas que no esten definidas van a este lugar */}
            <Route component={HomeScreen} />

            {/* Este es por si quieres redireccionar a un gar */}
          </Switch>
        </div>
      </div>
    </Router>
  );
};
