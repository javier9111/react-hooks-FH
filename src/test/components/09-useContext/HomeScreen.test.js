import React from "react";
import { shallow } from "enzyme";
import { HomeScreen } from "../../../component/09-useContext/HomeScreen";
import { UserContext } from "../../../component/09-useContext/UserContext";
describe("test on <HomeScreen/>", () => {
  const user = {
    name: "Javier Robles",
    email: "javier.robles@gmail.com",
  };

  //this simulates the usercontext. the UserContext.Provider, for complete test it must use mount.
  const wrapper = shallow(
    <UserContext.Provider
      value={{
        user,
      }}
    >
      <HomeScreen />
    </UserContext.Provider>
  );

  test("should show properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
