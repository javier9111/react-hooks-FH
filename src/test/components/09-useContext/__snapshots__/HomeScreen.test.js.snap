// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <HomeScreen/> should show properly 1`] = `<HomeScreen />`;
